src/op/nms.cc

```diff
@@ -300,13 +280,31 @@ int32_t nms::add_ipv4_cidr(const char *a_buf, uint32_t a_buf_len)
         // -------------------------------------------------
         uint32_t l_masked_addr = l_in.s_addr & l_mask;
         // -------------------------------------------------
-        // add
+        // create sets if not already created
         // -------------------------------------------------
-        if(!m_ipv4_mask_map)
+        if(!ipv4_arr)
         {
-                m_ipv4_mask_map = new ipv4_mask_map_t();
+                ipv4_arr = new ipv4_set_t[33];
         }
-        (*m_ipv4_mask_map)[l_bits].insert(l_masked_addr);
+        // ----------------------------------------
+        // iterate through to handle redundancies
+        // ----------------------------------------
+        for(uint32_t i = 0 ;i < 33; ++i) 
+        {
+                int32_t temp_mask = (i == 0) ? 0 : htonl(~((1 << (32 - i))-1 ));
+                // ----------------------------------------
+                // Check each prefix of current prefix is 
+                // already in nms
+                // ----------------------------------------
+                if(i < l_bits && ipv4_arr[i].find(l_masked_addr & temp_mask)!=ipv4_arr[i].end()) 
+                {
+                        return WAFLZ_STATUS_OK;
+                }
+        }
```

Comment:
I think we dont need this bit. Since we discussed the object is built from scratch and uniqueness can be handled at API level